core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -949,6 +949,93 @@ public void modeInUnsupportedFormat() {
+    assertEquals(nullValue(), eval(DSL.week_of_year(nullRef)));
+    assertEquals(missingValue(), eval(DSL.week_of_year(missingRef)));
+  }
+
+  @Test
+  public void invalid_week_of_year() {
+    testNullMissingWeekOfYear(DATE);
+    testNullMissingWeekOfYear(DATETIME);
+    testNullMissingWeekOfYear(TIMESTAMP);
+    testNullMissingWeekOfYear(STRING);
+
+    when(nullRef.type()).thenReturn(INTEGER);
+    when(missingRef.type()).thenReturn(INTEGER);
+    assertEquals(nullValue(), eval(DSL.week_of_year(DSL.literal("2019-01-05"), nullRef)));
+    assertEquals(missingValue(), eval(DSL.week_of_year(DSL.literal("2019-01-05"), missingRef)));
+
+    when(nullRef.type()).thenReturn(DATE);
+    when(missingRef.type()).thenReturn(INTEGER);
+    assertEquals(missingValue(), eval(DSL.week_of_year(nullRef, missingRef)));
+  }
+
+  @Test
+  public void week_of_year() {
+    testNullMissingWeekOfYear(DATE);
+    testNullMissingWeekOfYear(DATETIME);
+    testNullMissingWeekOfYear(TIMESTAMP);
+    testNullMissingWeekOfYear(STRING);
+
+    FunctionExpression expression = DSL
+        .week_of_year(DSL.literal(new ExprTimestampValue("2019-01-05 01:02:03")));
+    assertEquals(INTEGER, expression.type());
+    assertEquals("week_of_year(TIMESTAMP '2019-01-05 01:02:03')", expression.toString());
+    assertEquals(integerValue(0), eval(expression));
+
+    expression = DSL.week_of_year(DSL.literal("2019-01-05"));
+    assertEquals(INTEGER, expression.type());
+    assertEquals("week_of_year(\"2019-01-05\")", expression.toString());
+    assertEquals(integerValue(0), eval(expression));
+
+    expression = DSL.week_of_year(DSL.literal("2019-01-05 00:01:00"));
+    assertEquals(INTEGER, expression.type());
+    assertEquals("week_of_year(\"2019-01-05 00:01:00\")", expression.toString());
+    assertEquals(integerValue(0), eval(expression));
+
+    testWeekOfYear("2019-01-05", 0, 0);
+    testWeekOfYear("2019-01-05", 1, 1);
+    testWeekOfYear("2019-01-05", 2, 52);
+    testWeekOfYear("2019-01-05", 3, 1);
+    testWeekOfYear("2019-01-05", 4, 1);
+    testWeekOfYear("2019-01-05", 5, 0);
+    testWeekOfYear("2019-01-05", 6, 1);
+    testWeekOfYear("2019-01-05", 7, 53);
+
+    testWeekOfYear("2019-01-06", 0, 1);
+    testWeekOfYear("2019-01-06", 1, 1);
+    testWeekOfYear("2019-01-06", 2, 1);
+    testWeekOfYear("2019-01-06", 3, 1);
+    testWeekOfYear("2019-01-06", 4, 2);
+    testWeekOfYear("2019-01-06", 5, 0);
+    testWeekOfYear("2019-01-06", 6, 2);
+    testWeekOfYear("2019-01-06", 7, 53);
+
+    testWeekOfYear("2019-01-07", 0, 1);
+    testWeekOfYear("2019-01-07", 1, 2);
+    testWeekOfYear("2019-01-07", 2, 1);
+    testWeekOfYear("2019-01-07", 3, 2);
+    testWeekOfYear("2019-01-07", 4, 2);
+    testWeekOfYear("2019-01-07", 5, 1);
+    testWeekOfYear("2019-01-07", 6, 2);
+    testWeekOfYear("2019-01-07", 7, 1);
+
+    testWeekOfYear("2000-01-01", 0, 0);
+    testWeekOfYear("2000-01-01", 2, 52);
+    testWeekOfYear("1999-12-31", 0, 52);
```

Comment:
Can we add some tests for invalid dates + modes > 7 please. Would be useful to know how this implementation handles them (maybe returns null?).

Reply:
Addressed In https://github.com/Bit-Quill/opensearch-project-sql/pull/175/commits/b6c94ae6ceae47411508deb3f0c8a461fef4f700